ntp-proto/src/packet/v5/mod.rs

```diff
@@ -210,6 +218,49 @@ impl NtpHeaderV5 {
         }
     }
 
+    pub(crate) fn rate_limit_response(packet_from_client: Self) -> Self {
+        Self {
+            mode: NtpMode::Response,
+            flags: NtpFlags {
+                unknown_leap: false,
+                interleaved_mode: false,
+                status_message: true,
+            },
+            server_cookie: NtpServerCookie([b'R', b'A', b'T', b'E', 0, 0, 0, 0]),
+            client_cookie: packet_from_client.client_cookie,
+            poll: packet_from_client.poll.force_inc(),
+            ..Self::new()
+        }
```

Comment:
This is repeated a lot, maybe move the common parts into a function like `Self::kiss_response` and call it instead of `Self::new`

Reply:
done